modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java

```diff
@@ -86,6 +88,56 @@ public void mergeOpChangePrimaryKey() {
                 .check();
     }
 
+    @Test
+    public void batchWithConflictShouldBeRejectedEntirely() {
+        sql("CREATE TABLE test (id int primary key, val int)");
+
+        sql("INSERT INTO test values (1, 1)");
+
+        assertQuery("SELECT count(*) FROM test")
+                .returns(1L)
+                .check();
+
+        assertThrows(
```

Comment:
let's check that exception has right SQL code

Reply:
Also I think it should be SqlException instead of IgniteException

Reply:
fixed